Guida Utente (#116)
anitvam/pps-bunny: doc/report.md

del codice non correttamente funzionante, infatti se i test non passano, la pull request effettuata dallo sviluppatore
non viene mergiata.

Per quanto riguarda invece il package `it.unibo.pps.bunny.view` non sono stati implementati test di `ScalaTest` per
verificarne il funzionamento, in quanto si è deciso di effettuare prevalentemente del beta-testing, cioè di eseguire
direttamente l'applicativo per verificare ad esempio il movimento dei coniglietti e la gestione dei pannelli per l'
inserimento delle mutazioni e dei fattori, oltre che per avere una controprova visiva del corretto funzionamento delle
classi di model e dell'engine.

## Retrospettiva

## Guida utente

L'applicazione può essere lanciata in diversi modi:

* eseguendo il jar del proprio sistema operativo fornito nella sezione release di GitHub attraverso il
  comando `$ java -jar `
  path-to-downloaded-jar``
* scaricando il repository ed eseguendolo tramite `sbt` il comando `sbt run`
* importando il progetto ed eseguendolo nell'IDE che si preferisce

Una volta che l'applicativo è stato messo in esecuzione apparirà una board di gioco con l'aspetto riportato

> in [aggiungere screen applicazione] che rappresenta la GUI principale della simulazione con la quale si può interagire

Reviewer: l'ho caricata come GUI_screenshot, la puoi inserire

Una volta che l'applicativo è stato messo in esecuzione apparirà una board di gioco con l'aspetto riportato
in [aggiungere screen applicazione] che rappresenta la GUI principale della simulazione con la quale si può interagire
per avviarla. La board è formata da:

* un pannello principale posto in alto a sinistra nel quale avviene la simulazione vera e propria. Qui infatti si
  ritroveranno i coniglietti, l'orologio nell'angolo in alto a sinistra, i due bottoni per il cambiamento del clima in
  alto a destra e il bottone per l'aumento della velocità in basso a destra.
* un pannello multiplo, sottostante al precedente, che contiene i tre principali grafici:
    * il grafico della popolazione
    * il grafico delle proporzioni delle mutazioni suddivise per generazioni
    * il grafico dell'albero genealogico
* un pannello per l'aggiunta delle mutazioni in alto a destra con due bottoni per ciascuna mutazione, uno permette di introdurre la mutazione come dominante e l'altro come recessiva 
* un pannello per l'aggiunta di fattori disturbanti in basso a destra con una check box per ogni tipo di fattore che si
  può introdurre

> La simulazione parte quando si preme il pulsante "START", ma prima si possono comunque selezionare le mutazioni e i
> fattori che si trovano nei due rispettivi pannelli e ispezionare i grafici anche se ovviamente non ancora completi delle
> informazioni necessarie.

Reviewer 2: ```suggestion
La simulazione parte quando si preme il pulsante "START", ma prima si possono comunque selezionare le mutazioni e i
fattori che si trovano nei due rispettivi pannelli.
```

genotipo dei coniglietti solo durante la fase di riproduzione immediatamente successiva all'inserimento, come indicato
durante il gioco dalle scritte informative nel pannello. Similmente anche i fattori sono applicati, una volta aggiunti,
solo durante la rispettiva fase delle generazioni successive. Mentre i fattori possono essere inseriti e tolti a proprio
piacimento, le mutazioni no. Per quanto riguarda il clima e la velocità di esecuzione della simulazione, questi possono
essere variati dall'utente senza vincoli. Durante il corso della simulazione i vari grafici implementati saranno
popolati con le informazioni in tempo reale la velocità di esecuzione.

La simulazione termina o quando i coniglietti conquistano il mondo o quando si sono estinti, in entrambi i casi viene
mostrata una dialog finale con la descrizione dell'accaduto e si può ripartire da capo cliccando sul tasto restart che
comparirà al posto di quello dello speed up della velocità.

Quello che si consiglia è di non applicare inizialmente nessun fattore, ma di lasciare riprodurre per qualche
generazione i coniglietti. Questo consentirà di evitare la loro morte precoce e l'immeditata terminazione del gioco,

> permettendo alla simulazione di svilupparsi per un periodo di tempo maggiore.

Reviewer 2: toglierei questa parte

  può introdurre

La simulazione parte quando si preme il pulsante "START", ma prima si possono comunque selezionare le
mutazioni e i fattori che si trovano nei due rispettivi pannelli e ispezionare i grafici anche se ovviamente
non ancora completi delle informazioni necessarie.

Una volta avviata la simulazione si potrà aggiungere una mutazione in qualsiasi momento ma vista la logica della genetica verrà applicata di fatto alla generazione successiva, durante la riproduzione, come indicato durante il gioco dalle scritte informative nel pannello. Lo stesso discorso vale per l'introduzione di un fattore disturbante dal momento che ogni fattore a un suo preciso momento di applicazione, ovvero se si introducono i fattori dopo il loro periodo verrano applicati alla generazione successiva. Mentre i fattori possono essere inseririti e tolti a proprio piacimento, le mutazioni no, infatti per rimanere fedeli alla realtà una volta introdotta una mutazione non si potrà più rimuovere. Infine i grafici verranno popolati con le informazioni realtime e si potranno gestire il clima e
la velocità di esecuzione.

> Il simulatore termina quando i coniglietti conquistano il mondo oppure perchè si sono estinti, in entrambi i casi
> viene mostrata una dialog finale con la descrizione dell'accaduto e si può ripartire da capo cliccando sul tasto restart
> che comparirà al posto di quello dello speed up della velocità.

Reviewer 3: ```suggestion
La simulazione termina o quando i coniglietti conquistano il mondo o quando si sono estinti, in entrambi i casi
viene mostrata una dialog finale con la descrizione dell'accaduto e si può ripartire da capo cliccando sul tasto restart
che comparirà al posto di quello dello speed up della velocità.
```

Quello che si consiglia è di non applicare inizialmente nessun fattore, ma di lasciare riprodurre per qualche

> generazione i coniglietti. Questo consentirà di evitare la loro morte precoce e l'immeditata terminazione del gioco, permettendo alla simulazione di svilupparsi per un periodo di tempo maggiore. 

Reviewer 3: Non sono convinta di aggiungere questa parte

Per quanto riguarda invece il package `it.unibo.pps.bunny.view` non sono stati implementati test di `ScalaTest` per verificarne il funzionamento, in quanto si è deciso di effettuare prevalentemente del beta-testing, cioè di eseguire direttamente l'applicativo per verificare ad esempio il movimento dei coniglietti e la gestione dei pannelli per l'inserimento delle mutazioni e dei fattori, oltre che per avere una controprova visiva del corretto funzionamento delle classi di model e dell'engine.

## Retrospettiva

## Guida utente

L'applicazione può essere lanciata in diversi modi:

* tramite il jar del proprio sistema operativo fornito nella sezione release di GitHub sia con il comando `$ java -jar `

>   path-to-downloaded-jar`` che il doppio click diretto sul jar scariato

Reviewer: ```suggestion
  path-to-downloaded-jar`` che il doppio click diretto sul jar scaricato
```

* tramite `sbt` usando il comando `sbt run mainClass`
* importando il progetto ed eseguendolo nell'IDE che si preferisce

> Una volta messa in esecuzione apparirà una board di gioco, con l'aspetto riportato in [aggiungere screen applicazione], che rappresenta la GUI

Reviewer: ```suggestion
Una volta che l'applicativo è stato messo in esecuzione apparirà una board di gioco con l'aspetto riportato in [aggiungere screen applicazione] che rappresenta la GUI
```

principale della simulazione con la quale si può interagire per avviarla. La board è formata da:

* un pannello, il più grande dei quattro, posto in alto a sinistra nel quale avviene la simulazione vera e propria. Qui
  infatti si ritroveranno i coniglietti, l'orologio nell'angolo in alto a sinistra, i due bottoni per il cambiamento del
  clima in alto a destra e il bottone per l'aumento della velocità in basso a destra.

> * un pannello multiplo, sotto al pannello di gioco principale, in basso a sinistra che contiene i tre principali
>   grafici:

Reviewer: ```suggestion
* un pannello multiplo, sotto al pannello di gioco principale, che contiene i tre principali grafici:
```

  * il grafico della popolazione
  * il grafico delle proporzioni delle mutazioni suddivise per generazioni
  * il grafico dell'albero genealogico
* un pannello per l'aggiunta delle mutazioni in alto a destra con un bottone per ogni tipo di mutazione che si può
  introdurre
* un pannello per l'aggiunta di fattori disturbanti in basso a destra con una check box per ogni tipo di fattore che si
  può introdurre

Tutto comincia quando si preme il pulsante "START", ma nell'attesa di iniziare si possono comunque selezionare le

> mutazioni e i fattori che si trovano nei due rispettivi pannelli e si possono ispezionare i grafici anche se ovviamente

Reviewer: ```suggestion
mutazioni e i fattori che si trovano nei due rispettivi pannelli e ispezionare i grafici anche se ovviamente
```

non ancora completi delle informazioni necessarie.

> Una volta avviata la simulazione si potrà aggiungere una mutazione in qualsiasi momento ma vista la logica della
> genetica verrà applicata di fatto alla generazione successiva durante la riproduzione, come poi indicato durante il
> gioco dalle scritte d'informazione nel pannello. Stesso discorso vale per l'introduzione di un fattore distrurbante dal
> momento che ogni fattore a un suo preciso momento di applicazione, se si introducono i fattori dopo il loro periodo
> verrano applicati alla generazione successiva. Mentre per i fattori vale la regola che si possono inserire e togliere a
> proprio piacimento per le mutazioni no, infatti per rimanere fedeli alla realtà una volta introdotta una mutazione non
> si potrà più rimuovere. Infine i grafici verranno popolati con le informazioni realtime e si potranno gestire il clima e
> la velocità di esecuzione.

Reviewer: ```suggestion
Una volta avviata la simulazione si potrà aggiungere una mutazione in qualsiasi momento ma vista la logica della genetica verrà applicata di fatto alla generazione successiva, durante la riproduzione, come indicato durante il gioco dalle scritte informative nel pannello. Lo stesso discorso vale per l'introduzione di un fattore disturbante dal momento che ogni fattore a un suo preciso momento di applicazione, ovvero se si introducono i fattori dopo il loro periodo verrano applicati alla generazione successiva. Mentre i fattori possono essere inseririti e tolti a proprio piacimento, le mutazioni no, infatti per rimanere fedeli alla realtà una volta introdotta una mutazione non si potrà più rimuovere. Infine i grafici verranno popolati con le informazioni realtime e si potranno gestire il clima e
la velocità di esecuzione.
```

> Il simulatore termina o quando i coniglietti conquistano il mondo oppure perchè si sono estinti, in entrambi i casi

Reviewer: ```suggestion
Il simulatore termina quando i coniglietti conquistano il mondo oppure perchè si sono estinti, in entrambi i casi
```
